pkg/build/generator/generator.go

```diff
@@ -86,6 +92,29 @@ type streamRef struct {
 	tag string
 }
 
+// FetchServiceAccountSecrets retrieves the Secrets used for pushing and pulling
+// images from private Docker registries.
+func (g *BuildGenerator) FetchServiceAccountSecrets(namespace string) ([]kapi.Secret, error) {
+	// TODO: Check for the service account specified in the build config when it
+	//       will be added
+	if len(g.DefaultServiceAccountName) == 0 {
+		g.DefaultServiceAccountName = bootstrappolicy.BuilderServiceAccountName
+	}
+	var result []kapi.Secret
+	sa, err := g.ServiceAccounts.ServiceAccounts(namespace).Get(g.DefaultServiceAccountName)
+	if err != nil {
+		return result, err
```

Comment:
will the error that surfaces in this case make sense, given that the user isn't currently specifying a service account?